### PR TITLE
Parrotfish for Step Function, Exploration Stage

### PR DESCRIPTION
**Pull request by zyangbi**

### CODE REVIEW

File: src/logger.py [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 logging.config.dictConfig({"version": 1, "disable_existing_loggers": True})
 logging.basicConfig(
-    format="%(asctime)s - %(levelname)s - %(message)s", level=defaults.LOG_LEVEL
+    # format="%(asctime)s - %(levelname)s - %(message)s", level=defaults.LOG_LEVEL
```
Comment (arshiamoghimi) [inline src/logger.py:7]: remove commented code

File: src/main.py [hunk 2]
```diff
@@ -21,11 +23,14 @@ def main():
...
     )
+    parser.add_argument(
+        "--step-function", action="store_true", help="Optimize a step function"
+    )
 
     args = parser.parse_args()
 
     if args.verbose:
-        logger.setLevel(level=logging.INFO)
+        logger.setLevel(level=logging.DEBUG)
```
Comment (arshiamoghimi) [inline src/main.py:33]: Set to info for merging

File: src/main.py [hunk 3]
```diff
@@ -44,11 +52,17 @@ def main():
         logger.critical(f"No configuration file is found in {config_file_path}")
         exit(1)
 
-    parrotfish = Parrotfish(configuration)
+    if args.step_function:
+        step_function = StepFunction(configuration)
+        step_function.optimize()
+        pass
```
Comment (arshiamoghimi) [inline src/main.py:58]: I believe this is redundant. Remove if so

File: tests/step_function/test_step_function.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
+from unittest.mock import MagicMock, Mock, patch
+
+import pytest
+
+from src.exploration.aws.aws_config_manager import AWSConfigManager
+from src.step_function.states import Task, Parallel, Map, Workflow
+from src.step_function.step_function import StepFunction
+
+
+# from src.exploration.aws.aws_config_manager import AWSConfigManager
```
Comment (arshiamoghimi) [inline tests/step_function/test_step_function.py:10]: remove

File: README.md
Comment (arshiamoghimi) [inline README.md:1]: please list the supported state types for the step function mode and mention where someone can add support for more types in the future.

**Review by arshiamoghimi — CHANGES_REQUESTED** [review]
Thanks for the clean coding and adding tests. I have left some minor commnets

File: src/main.py (hunk 3 above)
Comment (zyangbi) [reply src/main.py:58]: This invokes StepFunction class, which handles the optimization of a step function.

**Review by arshiamoghimi — APPROVED** [review]

**Pull request merged**